src/plugins/visualizations/public/visualize_app/components/viz_chart_warning.tsx

```diff
@@ -8,32 +8,42 @@
 
 import React, { FC } from 'react';
 import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
 import { EuiCallOut, EuiLink } from '@elastic/eui';
 import { useKibana } from '@kbn/kibana-react-plugin/public';
 import { VisualizeServices } from '../types';
-import { CHARTS_WITHOUT_SMALL_MULTIPLES } from '../utils/split_chart_warning_helpers';
-import type { CHARTS_WITHOUT_SMALL_MULTIPLES as CHART_WITHOUT_SMALL_MULTIPLES } from '../utils/split_chart_warning_helpers';
+import {
+  CHARTS_WITHOUT_SMALL_MULTIPLES,
+  CHARTS_TO_BE_DEPRECATED,
+} from '../utils/split_chart_warning_helpers';
+import type {
+  CHARTS_WITHOUT_SMALL_MULTIPLES as CHART_WITHOUT_SMALL_MULTIPLES,
+  CHARTS_TO_BE_DEPRECATED as CHART_TO_BE_DEPRECATED,
+} from '../utils/split_chart_warning_helpers';
 
 interface Props {
-  chartType: CHART_WITHOUT_SMALL_MULTIPLES;
+  chartType: CHART_WITHOUT_SMALL_MULTIPLES | CHART_TO_BE_DEPRECATED;
   chartConfigToken: string;
+  mode?: 'old' | 'new';
 }
 
 interface WarningMessageProps {
   canEditAdvancedSettings: boolean | Readonly<{ [x: string]: boolean }>;
   advancedSettingsLink: string;
+  mode?: 'old' | 'new';
 }
 
 const SwitchToOldLibraryMessage: FC<WarningMessageProps> = ({
   canEditAdvancedSettings,
   advancedSettingsLink,
+  mode = 'old',
 }) => {
   return (
     <>
       {canEditAdvancedSettings && (
         <FormattedMessage
           id="visualizations.newChart.conditionalMessage.newLibrary"
-          defaultMessage="Switch to the old library in {link}"
+          defaultMessage="Switch to the {type} library in {link}"
```

Comment:
Will there ever be copy saying: "...Switch to the *old* library..."? I thought it's only displayed for when user uses depracated chart... 🤔

Reply:
Ok found it, sorry - when used for split chart aggregation 👌🏼

Reply:
This is a generic function. It is also used for displaying a warning for heatmap/gauges **new** library when they are using the small multiples (we still not support them for the new implementation. So this should be a variable because we want this to be old for this setting but new for the others ;)